### ISSUE TITLE
table parser needs further work

### ISSUE THREAD

**Issue by gdamore** [bytesparadise/libasciidoc#637]
In addition to #635 (styling for tables) it turns out that upon a bit more testing, the table parser is really limited.

It cannot deal with tables that use the [cols= ... ] syntax, it cannot deal with with tables that have extraneous blank lines, etc.

A little bit of testing with asciidoctor makes me believe that the table parser can be significantly improved by parsing them at a higher priority than other elements,   Basically once the table escape is started, the interior needs to be treated as elements, ignoring extra blank lines -- except that the interior of a cell may have interior blank lines to permit embedded asciidoctor markup.

Asciidoctor's table mark up is quite rich, and we have a lot to doo here.

**Comment by gdamore** [comment]
Nested table mark up might be challenging btw.

**Comment by xcoulon** [comment]
yes, as I said in https://github.com/bytesparadise/libasciidoc/issues/635#issuecomment-648388621, support for table is pretty basic for now :(
